External/FEXCore/include/FEXCore/Debug/InternalThreadState.h

```diff
@@ -112,8 +112,15 @@ namespace FEXCore::Core {
     std::shared_mutex ObjectCacheRefCounter{};
     bool DestroyedByParent{false};  // Should the parent destroy this thread, or it destory itself
 
-    alignas(16) FEXCore::Core::CpuStateFrame BaseFrameState{};
+    struct DeferredSignalState {
+      siginfo_t Info;
+      int Signal;
+    };
+
+    fextl::vector<DeferredSignalState> DeferredSignalFrames;
```

Comment:
Added a comment that the vector is treated as a FILO because async signals aren't guaranteed to be executed in any particular order.

Reply:
Does this mean we process asynchronous events in the reverse order of them being signaled? I guess that's done to avoid some minor overhead from erase-front-element operations on `std::vector`.

How does this trade off against chances of applications relying on signaling order? (You mentioned there are no guarantees, but I imagine there might be weak expectations like "this signal gets processed *in reasonable time* even if we spam other signals")

Reply:
(I think we discussed this externally already, but just to reiterate the new comment sounds well-fitting then. Leaving the thread here "unresolved" to keep this interesting background visible for future reference.)